services/mailer/mailer.go

```diff
@@ -147,42 +147,54 @@ type smtpSender struct{}
 func (s *smtpSender) Send(from string, to []string, msg io.WriterTo) error {
 	opts := setting.MailService
 
-	host, port, err := net.SplitHostPort(opts.Host)
-	if err != nil {
-		return err
+	var network string
+	var address string
+	if opts.Protocol == "smtp+unix" {
+		network = "unix"
+		address = opts.SmtpAddr
+	} else {
+		network = "tcp"
+		address = net.JoinHostPort(opts.SmtpAddr, opts.SmtpPort)
 	}
 
-	tlsconfig := &tls.Config{
-		InsecureSkipVerify: opts.SkipVerify,
-		ServerName:         host,
+	conn, err := net.Dial(network, address)
+	if err != nil {
+		return err
 	}
+	defer conn.Close()
 
-	if opts.UseCertificate {
-		cert, err := tls.LoadX509KeyPair(opts.CertFile, opts.KeyFile)
-		if err != nil {
-			return err
+	var tlsconfig *tls.Config
+	if opts.Protocol == "smtps" || opts.Protocol == "smtp+startls" {
+		tlsconfig = &tls.Config{
+			InsecureSkipVerify: opts.ForceTrustServerCert,
+			ServerName:         opts.SmtpAddr,
 		}
-		tlsconfig.Certificates = []tls.Certificate{cert}
-	}
 
-	conn, err := net.Dial("tcp", net.JoinHostPort(host, port))
-	if err != nil {
-		return err
+		if opts.UseClientCert {
+			cert, err := tls.LoadX509KeyPair(opts.ClientCertFile, opts.ClientKeyFile)
+			if err != nil {
+				return err
+			}
+			tlsconfig.Certificates = []tls.Certificate{cert}
+		}
 	}
-	defer conn.Close()
 
-	isSecureConn := opts.IsTLSEnabled || (strings.HasSuffix(port, "465"))
-	// Start TLS directly if the port ends with 465 (SMTPS protocol)
-	if isSecureConn {
+	if opts.Protocol == "smtps" {
 		conn = tls.Client(conn, tlsconfig)
 	}
 
+	var host string
+	if opts.Protocol == "smtp+unix" {
+		host = opts.SmtpAddr
+	} else {
+		host = "localhost"
+	}
```

Comment:
```suggestion
	host := "localhost"
	if opts.Protocol == "smtp+unix" {
		host = opts.SmtpAddr
	}
```

Reply:
I initially did this but wasn't sure what was idiomatic; will change to that version.